Documentation/articles/command/command_handler.md

```diff
@@ -0,0 +1,44 @@
+---
+title: Handling Commands
+description: Recommendations and Requirements when Handling a Command
+keywords: 
+author: einari, smithmx
+---
+
+# Command Handler
+
+## Background
+
+A *Command Handler* handles commands.  The command handler is responsible for executing the appropriate methods on Aggregate Roots and Domain Services to fulfill the intent of the *Command*.
+
+## Requirements
+
+A *Command Handler* must implement the *IHandleCommands* interface.  This is a simple marker interface with no methods to implement.
+
+To handle a *Command*, you create a method called *Handle* with a single parameter that is the *Command* you wish to handle.  A *Command Handler* does not return anything.  It is not permitted to return a value from a *Command*.  When a *Command* is executed, you will receive a *CommandResult*
+which will indicate whether the *Command* succeeded and, in the case of failure, why the *Command* failed.  No artifacts that were created or updated as a result of the *Command* will be present on the *CommandResult*.
```

Comment:
We shouldn't use the term "artifact" here, we are using that for another purpose now and it could be confusing.